exotic.py

```diff
@@ -407,30 +410,32 @@ def user_input(prompt, type_, val1=None, val2=None):
             return option
 
 
+# Check if user's directory contains imaging files that are able to be reduced
 def check_file_extensions(directory, fileName):
-    # Add / to end of directory if user does not input it
-    if directory[-1] != "/":
-        directory += "/"
-
     # Find fits files
-    fits_extensions = ["*.FITS", "*.FIT", "*.fits", "*.fit"]
-    # Loop until we find something
-    looper = True
-    while looper:
-        for exti in fits_extensions:
-            inputfiles = g.glob(directory + exti)
-            # If we find files, then stop the for loop and while loop
-            if len(inputfiles) > 0:
-                looper = False
-                break
-        # If we don't find any files, then we need the user to check their directory and loop over again....
-        if len(inputfiles) == 0:
-            print("Error: " + fileName + " .FITS files not found in " + directory + ". Please try again.")
-            directory = str(input("Enter the Directory Path where FITS Image Files are located: "))
+    fits_extensions = ['.FITS', '.FIT', '.FTS', '.fits', '.fit', '.fts']
+    inputfiles = []
+
+    while True:
+        try:
             # Add / to end of directory if user does not input it
             if directory[-1] != "/":
                 directory += "/"
-    return directory, inputfiles
+
+            # Loop until we find something
+            for exti in fits_extensions:
+                for file in os.listdir(directory):
+                    if file.endswith(exti):
+                        inputfiles.append(os.path.join(directory, file))
+                # If we find files, then stop the for loop and while loop
+                if inputfiles:
+                    return directory, inputfiles
+
+            # If we don't find any files, then we need the user to check their directory and loop over again....
+            raise FileNotFoundError
+        except FileNotFoundError:
+            print("Error: " + fileName + " .FITS or .FTS files not found in " + directory + ". Please try again.")
+            directory = str(input("Enter the Directory Path where FITS or .FTS Image Files are located: "))
```

Comment:
In the future, we should ask the user to input their file extension if EXOTIC cannot automatically find it. They might use a non-standard image extension.

Reply:
I will definitely modify this portion of the code later on for accepting/searching those extensions when solving the ticket - I added this as a bandaid for those that do not want convert between .fits and .fts to run in EXOTIC. I can revert this back though and solve it at a later time if needed to complete issue #34

Reply:
I would change the matching algorithm slightly. 
```
   fits_extensions = ['.fits', '.fit', '.fts']
   ...
   if file.lower().endswith(exti.lower()): 
       ...
```
Now there is a smaller list of extensions to manage, and as well the users' files may be named with `.Fits` or `.Fit` or any case permutation.

Reply:
This probably follows up to other code segments, too. After the FileNotFoundError (and any other blocks) it would be helpful to have a fall through exception, like `OSError` to catch any local I/O Error that may occur. [According to the latest Python docs, `OSError` is basically at the root of file system failures.](https://docs.python.org/3/library/exceptions.html#OSError)

Reply:
Updated with your suggestion John to the following:

`
              if os.path.isdir(directory):
                 # Loop until we find something
                 for exti in file_extensions:
                     for file in os.listdir(directory):
                         if file.lower().endswith(exti.lower()):
                             inputfiles.append(os.path.join(directory, file))
                     # If we find files, then stop the for loop and while loop
                     if inputfiles:
                         return directory, inputfiles

                  # If we don't find any files, then we need the user to check their directory and loop over again
                 if not inputfiles:
                     raise FileNotFoundError

              # If the directory does not exist
             else:
                 raise OSError

          except FileNotFoundError:
             print("Error: " + fileName + " files not found with .fits, .fit or .fts extensions in " + directory + ". Please try again.")
             directory = input("Enter the directory path where " + fileName + " files are located: ")
         except OSError:
             print("Error: No such directory exists. Please try again.")
             directory = input("Enter the directory path where " + fileName + " files are located: ")
`

Reply:
It may be good to ask if they would like to create the directory if it doesn't exist.